modin/pandas/series.py

```diff
@@ -1031,7 +1031,14 @@ def eq(
         Return Equal to of series and `other`, element-wise (binary operator `eq`).
         """
         new_self, new_other = self._prepare_inter_op(other)
-        return super(Series, new_self).eq(new_other, level=level, axis=axis)
+        return new_self._binary_op(
+            "eq",
+            new_other,
+            level=level,
+            fill_value=fill_value,
+            axis=axis,
+            squeeze_other=isinstance(other, (pandas.Series, Series)),
```

Comment:
```suggestion
            squeeze_other=isinstance(other, Series),
```

Reply:
IIRC basically the logic was written taking into account that pandas types are not passed directly to methods.